### PR TITLE
Remove enkf_plot_gendata and genvector

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/clib/lib/python/enkf_fs_general_data.cpp [hunk 1]
```diff
@@ -7,26 +8,22 @@
...
+           const std::vector<int> &realizations, int report_step) {
             const int data_size =
-                enkf_plot_gendata_get_data_size(enkf_plot_gendata);
-            if (data_size < 0) {
-                throw pybind11::value_error(
-                    "No data has been loaded for report step");
-            }
-            auto active_mask = enkf_plot_gendata_active_mask(enkf_plot_gendata);
-            const int realization_size = std::size(realizations);
+                enkf_config_node_get_data_size(config_node, report_step);
```
Comment (oyvindeide) [inline src/clib/lib/python/enkf_fs_general_data.cpp:14]: It fails here, I am not entirely sure why

**Comment by oyvindeide** [comment]
test this please

**Comment by oyvindeide** [comment]
test ert please

File: src/clib/lib/python/enkf_fs_summary_data.cpp [hunk 2]
```diff
@@ -1,9 +1,7 @@
 #include <ert/enkf/enkf_plot_tvector.hpp>
+#include <ert/enkf/ensemble_config.hpp>
```
Comment (pinkwah) [inline src/clib/lib/python/enkf_fs_summary_data.cpp:2]: Why?

**Review by pinkwah — APPROVED** [review]
:tada:

File: src/clib/lib/python/enkf_fs_summary_data.cpp (hunk 2 above)
Comment (oyvindeide) [reply src/clib/lib/python/enkf_fs_summary_data.cpp:2]: This for example: `Cwrap<ensemble_config_type> ensemble_config`

**Pull request merged**